### PR TITLE
Add optional SPF record and custom FROM domain

### PR DESCRIPTION
**Pull request by joe-niland**
## what

- Add ability to create an SPF record for the domain identity
- Add ability to create a custom FROM subdomain for the domain entity
- Increase TTL on DKIM and SES verification records in line with recommendations from DNS service providers (e.g. https://www.digicert.com/faq/dns/what-is-ttl)

## why

- An SPF record improves email deliverability and makes it more difficult to use the domain in an email spoofing attempt.
- Sending mail from a subdomain of the verified domain improves SPAM score (HEADER_FROM_DIFFERENT_DOMAINS)
- Reduce DNS query volume on mainly static records

## references

None

### CODE REVIEW

File: .github/renovate.json
Comment (hans-d) [inline .github/renovate.json:1]: Fix from the Make dance, or manually done? In the case of the latter, would keep this as a sperate PR.

File: examples/complete/main.tf
Comment (hans-d) [inline examples/complete/main.tf:1]: If possible, please split the "fixing of terratest" from the actual change / addition needed for this change

File: .github/renovate.json
Comment (joe-niland) [reply .github/renovate.json:1]: Yes it was from make.. wanted to make sure it was up to date as this repo hasn't been touched for a while.

Another PR sounds good - to be merged before this one.
But I saw somewhere that there was a new way of managing these workflows. Do you know what it is?

File: examples/complete/main.tf
Comment (joe-niland) [reply examples/complete/main.tf:1]: Isn't it better to align the tests with the new vars? Or am I misunderstanding your point?

File: .github/renovate.json
Comment (hans-d) [reply .github/renovate.json:1]: stuff is in progress / in various stages / aimed at various aspects. So life will become easier, but currently need to use the current rain dance procedure.
Splitting it a bit would be appreciated (bite size improvements)

File: examples/complete/main.tf
Comment (hans-d) [reply examples/complete/main.tf:1]: I see a combination of "fiding up the test - eg newer versions, change of cidr variable - that are not directly related to the spf bit.
If we can get those fixes in separately, than here we can just do the additional var adding, again making life easier.

Comment (hans-d) [reply examples/complete/main.tf:1]: (quite ok to combine the readme/gha update with this fixing the test)

**Comment by joe-niland** [comment]
@hans-d thanks - changes made as requested

**Comment by joe-niland** [comment]
@Mergifyio refresh

**Review by milldr — APPROVED** [review]

**Pull request merged**